### ISSUE TITLE
making trainer dependent on evaler

### ISSUE THREAD

**Issue by datavizweb** [tensorflow/lingvo#81]
** General question about lingvo **

* Trainer, Evaler run in different process independently of one another. How trainer knows that it has to reduce the learning rate after few steps/epoch. For Adam with constant learning rate or learning rate schedule like transform Independent trainer, evaler is okay to have. But for new-bob can they be made dependent?
* Running trainer,evaler in same job will that make them dependent and is there way to change the optimization parameter after few steps/epoch based on evaler ?

The paper talks about trainer, evaler where in they seems to run independent of other.

** https://arxiv.org/pdf/1902.08295.pdf **

After the variables are initialized, the Trainer will run training, i.e. calling
session . run with the model’s train_op, in a loop, and the Controller will
produce training summaries for each step. When enough steps have passed,
the Controller writes a new checkpoint to the log directory. The Evaler and
Decoder detect this new checkpoint and evaluates the model at that checkpoint
to generate summaries. This process then continues until the user terminates all
jobs or p. train . max_steps is reached.

**Comment by jonathanasdf** [comment]
Yes, they are run as independent jobs. See https://github.com/tensorflow/lingvo/blob/24d82bd643d2d4ff4f478843458fe4de7e93fa41/lingvo/core/lr_schedule.py#L506 for adjusting learning rate based on evaler results (read from the tfevents output files).

**Comment by datavizweb** [comment]
Thanks, I will have a look at that.

**Comment by zh794390558** [comment]
watch this.
